### PR TITLE
DDCE-2729 - removed reactive mongo, updated dependencies + clean up

### PR DESCRIPTION
**Pull request by ina-tu**
- removed reactive-mongo dependency
- upgraded sbt version from 1.3.13 to 1.6.2 (missed out during sbt version upgrades)
- upgraded some dependencies and addressed some scalastyle warnings
- added `run_all_tests.sh` script
- updated README.md

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -71,3 +71,5 @@ lazy val testSettings: Seq[Def.Setting[_]] = Seq(
 )
 
 dependencyOverrides ++= AppDependencies.overrides
```
Comment (frikit) [inline build.sbt:73]: please delete overrides

File: conf/application.conf [hunk 2]
```diff
@@ -31,7 +31,6 @@ play.modules.enabled += "uk.gov.hmrc.play.bootstrap.HttpClientModule"
 play.modules.enabled += "uk.gov.hmrc.play.bootstrap.AuditModule"
```
Comment (frikit) [inline conf/application.conf:31]: please change uk.gov.hmrc.play.bootstrap.AuditModule to uk.gov.hmrc.play.audit.AuditModule

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -3,30 +3,26 @@ import sbt._
...
 
-  val compile: Seq[ModuleID] = Seq(
+  private val compile: Seq[ModuleID] = Seq(
     play.sbt.PlayImport.ws,
-    "org.reactivemongo"   %% "play2-reactivemongo"            % "1.1.0-play28-RC4",
     "uk.gov.hmrc"         %% "play-frontend-hmrc"             % "3.22.0-play-28",
     "uk.gov.hmrc"         %% "play-conditional-form-mapping"  % "1.11.0-play-28",
     "uk.gov.hmrc"         %% "domain"                         % "8.1.0-play-28",
-    "uk.gov.hmrc"         %% "bootstrap-frontend-play-28"     % "6.3.0",
+    "uk.gov.hmrc"         %% "bootstrap-frontend-play-28"     % "7.8.0",
```
Comment (frikit) [inline project/AppDependencies.scala:11]: can be updated

File: test/repositories/SubmissionSetFactorySpec.scala [hunk 4]
```diff
@@ -19,8 +19,7 @@ package repositories
 import base.SpecBase
 import generators.ModelGenerators
 import models.{RegistrationSubmission, UserAnswers, YearReturnType}
-import org.mockito.Matchers.any
-import org.mockito.Mockito.when
+import org.mockito.ArgumentMatchers.any
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 import play.api.libs.json.{JsNull, Json}
 import play.twirl.api.Html
```
Comment (frikit) [inline test/repositories/SubmissionSetFactorySpec.scala:25]: optimize imports here

File: test/repositories/SubmissionSetFactorySpec.scala [hunk 5]
```diff
@@ -33,11 +32,11 @@ import scala.collection.immutable.Nil
...
 
-  val mockCheckYourAnswersHelper: CheckYourAnswersHelper = mock[CheckYourAnswersHelper]
-  val mockTaxLiabilityService: TaxLiabilityService = mock[TaxLiabilityService]
-  val factory = new SubmissionSetFactory(mockCheckYourAnswersHelper, mockTaxLiabilityService)
+  private val mockCheckYourAnswersHelper: CheckYourAnswersHelper = mock[CheckYourAnswersHelper]
+  private val mockTaxLiabilityService: TaxLiabilityService = mock[TaxLiabilityService]
+  private val factory = new SubmissionSetFactory(mockCheckYourAnswersHelper, mockTaxLiabilityService)
 
-  implicit val hc: HeaderCarrier = HeaderCarrier()
+  private implicit val hc: HeaderCarrier = HeaderCarrier()
```
Comment (frikit) [inline test/repositories/SubmissionSetFactorySpec.scala:39]: seems like not used

**Review by frikit — CHANGES_REQUESTED** [review]
- Please fix sbt warning on line 61 and 63
- Please delete 43-47 lines in build.sbt
- Please enable license header in scalastyle.xml file
- Please update scalastyle xml license header
- Please fix this warning:
```
[warn] /Users/frikit/IdeaProjects/register-trust-tax-liability-frontend/test/generators/Generators.scala:67:14: method formatted in class StringFormat is deprecated (since 2.12.16): Use `formatString.format(value)` instead of `value.formatted(formatString)`,
[warn] or use the `f""` string interpolator. In Java 15 and later, `formatted` resolves to the new method in String which has reversed parameters.
[warn]       .map(_.formatted("%f"))
[warn]              ^
```
- Please fix this warning:
```
[warn] /Users/frikit/IdeaProjects/register-trust-tax-liability-frontend/test/models/EnumerableSpec.scala:52:65: method convertRightProjectionToValuable in trait EitherValues is deprecated: The .right.value syntax on Either has been deprecated and will be removed in a future version of ScalaTest. Please use .value instead.
[warn]           Json.fromJson[Foo](JsString(value.toString)).asEither.right.value mustEqual value
[warn]                      
```

File: build.sbt [hunk 6]
```diff
@@ -31,10 +31,10 @@ lazy val root = (project in file("."))
       "controllers.routes._"
     ),
     PlayKeys.playDefaultPort := 8838,
-    ScoverageKeys.coverageExcludedFiles := "<empty>;Reverse.*;.*filters.*;.*handlers.*;.*components.*;.*repositories.*;" +
+    ScoverageKeys.coverageExcludedFiles := "<empty>;Reverse.*;.*filters.*;.*handlers.*;.*components.*;" +
       ".*BuildInfo.*;.*javascript.*;.*FrontendAuditConnector.*;.*Routes.*;.*GuiceInjector;" +
       ".*ControllerConfiguration;.*LanguageSwitchController",
-    ScoverageKeys.coverageMinimumStmtTotal := 70,
+    ScoverageKeys.coverageMinimumStmtTotal := 80,
```
[the remark this answers is not in the document]
Comment (ina-tu) [reply build.sbt]: done

File: test/repositories/SubmissionSetFactorySpec.scala (hunk 5 above)
Comment (ina-tu) [reply test/repositories/SubmissionSetFactorySpec.scala]: done

File: project/AppDependencies.scala (hunk 3 above)
Comment (ina-tu) [reply project/AppDependencies.scala]: done

File: project/AppDependencies.scala [hunk 7]
```diff
@@ -3,30 +3,26 @@ import sbt._
 
 object AppDependencies {
 
-  val compile: Seq[ModuleID] = Seq(
+  private val compile: Seq[ModuleID] = Seq(
     play.sbt.PlayImport.ws,
-    "org.reactivemongo"   %% "play2-reactivemongo"            % "1.1.0-play28-RC4",
     "uk.gov.hmrc"         %% "play-frontend-hmrc"             % "3.22.0-play-28",
     "uk.gov.hmrc"         %% "play-conditional-form-mapping"  % "1.11.0-play-28",
```
[the remark this answers is not in the document]
Comment (ina-tu) [reply project/AppDependencies.scala]: done

File: conf/application.conf (hunk 2 above)
Comment (ina-tu) [reply conf/application.conf]: done

File: build.sbt (hunk 1 above)
Comment (ina-tu) [reply build.sbt]: done

**Review by frikit — APPROVED** [review]

**Pull request merged**